cmd/manager/main.go

```diff
@@ -112,120 +137,117 @@ func main() {
-	// Setup Scheme for all defined resources
-	if err := apis.AddToScheme(mgr.GetScheme()); err != nil {
-		logrus.Fatal(err)
-	}
-	// Add external resource to scheme
-	if err := olmv1alpha1.AddToScheme(mgr.GetScheme()); err != nil {
-		logrus.Fatal(err)
-	}
-	if err := v1beta1.AddToScheme(mgr.GetScheme()); err != nil {
-		logrus.Fatal(err)
-	}
-	// If the config API is available add the config resources to the scheme
-	if configv1.IsAPIAvailable() {
-		if err := apiconfigv1.AddToScheme(mgr.GetScheme()); err != nil {
-			logrus.Fatal(err)
-		}
+		logger.Fatal(err)
 	}
 
-	stopCh := signals.Context().Done()
-
-	var statusReporter status.Reporter = &status.NoOpReporter{}
-	if clusterOperatorName != "" {
-		statusReporter, err = status.NewReporter(cfg, mgr, namespace, clusterOperatorName, os.Getenv("RELEASE_VERSION"), stopCh)
-		if err != nil {
-			logrus.Fatal(err)
-		}
-	}
-
-	// Populate the global default OperatorSources definition and config
-	err = defaults.PopulateGlobals()
-	if err != nil {
-		logrus.Fatal(err)
-	}
-
-	// Setup all Controllers
-	if err := controller.AddToManager(mgr, options.ControllerOptions{}); err != nil {
-		logrus.Fatal(err)
-	}
-
-	logrus.Info("Setting up health checks")
+	logger.Info("setting up health checks")
 	http.HandleFunc("/healthz", func(w http.ResponseWriter, r *http.Request) {
 		w.WriteHeader(http.StatusOK)
 	})
 	go http.ListenAndServe(":8080", nil)
 
-	// Wait until this instance becomes the leader.
-	logrus.Info("Waiting to become leader.")
-	err = leader.Become(context.TODO(), leaderElectionConfigMapName)
-	if err != nil {
-		logrus.Fatal(err, "Failed to retry for leader lock")
-	}
-	logrus.Info("Elected leader.")
+	ctx, cancel := context.WithCancel(signals.Context())
+	defer cancel()
+
+	run := func(ctx context.Context) {
+		stopCh := ctx.Done()
+		logger.Info("registering components")
+		var statusReporter status.Reporter = &status.NoOpReporter{}
+		if clusterOperatorName != "" {
+			logger.Info("setting up the marketplace clusteroperator status reporter")
+			statusReporter, err = status.NewReporter(cfg, mgr, namespace, clusterOperatorName, os.Getenv("RELEASE_VERSION"), stopCh)
+			if err != nil {
+				logger.Fatal(err)
+			}
+		}
+
+		// Populate the global default OperatorSources definition and config
+		if err := defaults.PopulateGlobals(); err != nil {
+			logger.Fatal(err)
+		}
 
-	logrus.Info("Starting the Cmd.")
+		logger.Info("setting up controllers")
+		if err := controller.AddToManager(mgr, options.ControllerOptions{}); err != nil {
+			logger.Fatal(err)
+		}
 
-	// migrate away from Marketplace API
-	clientGo, err := client.New(cfg, client.Options{Scheme: mgr.GetScheme()})
-	if err != nil && !k8sErrors.IsNotFound(err) {
-		logrus.Fatal(err, "Failed to instantiate the client for migrator")
-	}
-	migrator := migrator.New(clientGo)
-	err = migrator.Migrate()
-	if err != nil {
-		logrus.Error(err, "[migration] Error while migrating Marketplace away from OperatorSource API")
-	}
```

Comment:
Quick note to reviewers: the master/release-4.9 PR automatic cherrypick failed when attempting to apply these commit(s) to the existing cmd/manager/main.go implementation. As a result, I added the commit for removing the migration logic that was added during the 4.9 release timeframe: https://github.com/operator-framework/operator-marketplace/pull/421/commits/bcb928902743bf8f7942d721049be34372d6a659. Should be a safe change to sneak in given that OperatorSource APIs were deprecated in 4.5 and removed in 4.6+.